neo4j/session_with_context_test.go

```diff
@@ -569,7 +570,82 @@ func TestSession(st *testing.T) {
 		})
 	})
 
-	st.Run("Close", func(ct *testing.T) {
+	outer.Run("GetServerInfo", func(inner *testing.T) {
+
+		inner.Run("Retrieves info from first borrowed connection", func(t *testing.T) {
```

Comment:
We could test also the case where the connection already exists.

Reply:
since we mock the entire pool here, we cannot really make that distinction.
it's at the pool level that we can test such things, not at the session level